async-nats/src/header.rs

```diff
@@ -309,6 +309,10 @@ impl FromStr for HeaderName {
     type Err = ParseError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if s.contains(|c: char| !c.is_ascii_alphanumeric() && c != '-') {
```

Comment:
If we want to use this error, it really should return some context why parsing failed.

Reply:
Error type could be changed into a `ParseHeaderNameError` and vice versa for header values, but follow up imo to unblock the rest of the queue.